### PR TITLE
feature: Jest matchers core

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Initial PR laying the groundwork for migration of Jest Native matchers to RNTL repo. Other matchers will be covered in separate PR so that the review process can be done properly for each matcher.

These matchers are not enable by default but could be added by doing import into `matchers/extend-expect`.

### Test plan

All tests should pass.

### CODE REVIEW

File: src/helpers/component-tree.ts [hunk 1]
```diff
@@ -92,3 +92,22 @@ export function getHostSiblings(
...
   );
 }
+
+/**
+ * Returns the unsafe root element of the tree (probably composite).
+ *
+ * @param element The element start traversing from.
+ * @returns The root element of the tree (host or composite).
+ */
+export function getRootElement(element: ReactTestInstance | null) {
```
Comment (thymikee) [inline src/helpers/component-tree.ts:102]: if it's unsafe, maybe let's state this explicitly with `getHostOrCompositeRootElement` or `unsafe_getRootElement`?

Comment (mdjastrzebski) [reply src/helpers/component-tree.ts:102]: it's internal fn, none of functions in this file are exported publically, so it's safe for us ;-)
Anyway I'll go for `getUnsafeRootElement()`

**Review by thymikee — APPROVED** [review]

File: src/helpers/component-tree.ts (hunk 1 above)
Comment (thymikee) [reply src/helpers/component-tree.ts:102]: I believe it should raise awareness in contributors as well :)

**Comment by mdjastrzebski** [comment]
🎉 Release in version: [v12.4.0](https://github.com/callstack/react-native-testing-library/releases/tag/v12.4.0)

**Pull request merged**